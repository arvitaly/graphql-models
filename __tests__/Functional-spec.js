"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments)).next());
    });
};
// tslint:disable:no-string-literal arrow-parens
const graphql_1 = require("graphql");
const graphql_relay_1 = require("graphql-relay");
const __1 = require("./..");
const collection1_1 = require("./../__fixtures__/collection1");
const data_1 = require("./../__fixtures__/data");
const animalId1 = graphql_relay_1.toGlobalId("Animal", "1");
fdescribe("Functional tests", () => {
    let adapter;
    let resolver;
    let schema;
    let collection;
    let graphqlSchema;
    beforeEach(() => {
        adapter = new data_1.DataAdapter();
        resolver = new __1.Resolver(adapter, data_1.callbacks, data_1.publisher);
        schema = new __1.Schema(resolver);
        collection = new __1.Collection([collection1_1.animalModel, collection1_1.postModel, collection1_1.userModel], {
            interfaces: [schema.getNodeDefinition().nodeInterface],
            resolveFn: resolver.resolve.bind(resolver),
        });
        schema.setCollection(collection);
        resolver.setCollection(collection);
        graphqlSchema = schema.getGraphQLSchema();
    });
    it("node", () => __awaiter(this, void 0, void 0, function* () {
        const result = yield graphql_1.graphql(graphqlSchema, `query Q1{  
            node(id:"${animalId1}"){
                ... on Animal{
                    name
                }
            }
        }`);
        if (result.errors) {
            result.errors.map((e) => {
                console.error(e);
                console.error(e.stack);
            });
        }
        expect(result).toMatchSnapshot();
    }));
    it("query one", () => __awaiter(this, void 0, void 0, function* () {
        const result = yield graphql_1.graphql(graphqlSchema, `query Q1{  
            viewer{
                animal(id:"${animalId1}"){
                    id
                    name
                    age
                    birthday
                    Weight
                    isCat
                }
            }
        }`);
        if (result.errors) {
            result.errors.map((e) => {
                console.error(e);
                console.error(e.stack);
            });
        }
        expect(result).toMatchSnapshot();
    }));
    it("query connection", () => __awaiter(this, void 0, void 0, function* () {
        const result = yield graphql_1.graphql(graphqlSchema, `query Q1{
            viewer{            
                animals(where:{nameContains:"x"}){
                    edges{
                        node{
                            ... on Animal{
                                name
                            }
                        }
                    }
                }
            }
        }`);
        expect(result).toMatchSnapshot();
    }));
    it("subscribe one", () => __awaiter(this, void 0, void 0, function* () {
        const subscriptionId = "123";
        const result = yield graphql_1.graphql(graphqlSchema, `query Q1{  
            viewer{
                animal(id:"${animalId1}"){
                    name
                }
            }
        }`, {}, {
            subscriptionId,
        });
        if (result.errors) {
            result.errors.map((e) => {
                console.error(e);
                console.error(e.stack);
            });
        }
        expect(result).toMatchSnapshot();
        const publishUpdateSpy = spyOn(data_1.publisher, "publishUpdate");
        adapter.update("animal", 1, { name: "testn" });
        expect(publishUpdateSpy.calls.allArgs()).toMatchSnapshot();
    }));
    it("subscribe connection", () => __awaiter(this, void 0, void 0, function* () {
        const subscriptionId = "123";
        const result = yield graphql_1.graphql(graphqlSchema, `query Q1{
            viewer{            
                animals(where:{nameContains:"x"}){
                    edges{
                        node{
                            ... on Animal{
                                name
                            }
                        }
                    }
                }
            }
        }`, {}, {
            subscriptionId,
        });
        if (result.errors) {
            result.errors.map((e) => {
                console.error(e);
                console.error(e.stack);
            });
        }
        expect(result).toMatchSnapshot();
        const publishUpdateSpy = spyOn(data_1.publisher, "publishUpdate");
        adapter.update("animal", 1, { name: "testn" });
        expect(publishUpdateSpy.calls.allArgs()).toMatchSnapshot();
        const publishAddSpy = spyOn(data_1.publisher, "publishAdd");
        adapter.create("animal", { name: "y" });
        expect(publishAddSpy.calls.allArgs()).toMatchSnapshot();
        adapter.create("animal", { name: "axz" });
        expect(publishAddSpy.calls.allArgs()).toMatchSnapshot();
    }));
});
//# sourceMappingURL=Functional-spec.js.map