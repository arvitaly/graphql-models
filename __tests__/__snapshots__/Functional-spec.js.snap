exports[`Functional tests mutation create 1`] = `
Object {
  "data": Object {
    "createPost": Object {
      "post": Object {
        "animals": Object {
          "edges": Array [
            Object {
              "node": Object {
                "Weight": 1008.1,
                "age": 1008,
                "birthday": "Wed, 07 Dec 2016 10:42:48 GMT",
                "id": "QW5pbWFsOjg=",
                "isCat": true,
                "name": "animal1",
              },
            },
            Object {
              "node": Object {
                "Weight": 1009.1,
                "age": 1009,
                "birthday": "Wed, 10 Nov 2100 17:00:00 GMT",
                "id": "QW5pbWFsOjk=",
                "isCat": false,
                "name": "animal2",
              },
            },
          ],
        },
        "owner": Object {
          "name": "user5",
          "pets": Object {
            "edges": Array [
              Object {
                "node": Object {
                  "Weight": 1007.1,
                  "age": 1007,
                  "birthday": "Wed, 10 Nov 2100 17:00:00 GMT",
                  "id": "QW5pbWFsOjc=",
                  "isCat": false,
                  "name": "pet1",
                },
              },
            ],
          },
        },
      },
    },
  },
}
`;

exports[`Functional tests node 1`] = `
Object {
  "data": Object {
    "node": Object {
      "name": "Rex",
    },
  },
}
`;

exports[`Functional tests query connection 1`] = `
Object {
  "data": Object {
    "viewer": Object {
      "animals": Object {
        "edges": Array [
          Object {
            "node": Object {
              "name": "Rex",
            },
          },
          Object {
            "node": Object {
              "name": "Felix",
            },
          },
        ],
      },
    },
  },
}
`;

exports[`Functional tests query one: animal 1`] = `
Object {
  "data": Object {
    "viewer": Object {
      "animal": Object {
        "Weight": 10.1,
        "age": 6,
        "birthday": "Wed, 10 Nov 2010 17:00:00 GMT",
        "id": "QW5pbWFsOjE=",
        "isCat": false,
        "name": "Rex",
      },
    },
  },
}
`;

exports[`Functional tests query one: post 1`] = `
Object {
  "data": Object {
    "viewer": Object {
      "post": Object {
        "animals": Object {
          "edges": Array [
            Object {
              "node": Object {
                "Weight": 10.1,
                "age": 6,
                "birthday": "Wed, 10 Nov 2010 17:00:00 GMT",
                "id": "QW5pbWFsOjE=",
                "isCat": false,
                "name": "Rex",
              },
            },
            Object {
              "node": Object {
                "Weight": 13.1,
                "age": 3,
                "birthday": "Thu, 28 Feb 2013 17:00:00 GMT",
                "id": "QW5pbWFsOjI=",
                "isCat": false,
                "name": "Felix",
              },
            },
            Object {
              "node": Object {
                "Weight": 2.1,
                "age": 8,
                "birthday": "Mon, 05 May 2008 17:00:00 GMT",
                "id": "QW5pbWFsOjM=",
                "isCat": false,
                "name": "Jeck",
              },
            },
          ],
        },
        "owner": Object {
          "name": "John",
          "pets": Object {
            "edges": Array [
              Object {
                "node": Object {
                  "Weight": 13.1,
                  "age": 3,
                  "birthday": "Thu, 28 Feb 2013 17:00:00 GMT",
                  "id": "QW5pbWFsOjI=",
                  "isCat": false,
                  "name": "Felix",
                },
              },
              Object {
                "node": Object {
                  "Weight": 2.1,
                  "age": 8,
                  "birthday": "Mon, 05 May 2008 17:00:00 GMT",
                  "id": "QW5pbWFsOjM=",
                  "isCat": false,
                  "name": "Jeck",
                },
              },
              Object {
                "node": Object {
                  "Weight": 12.1,
                  "age": 7,
                  "birthday": "Fri, 11 Dec 2009 17:00:00 GMT",
                  "id": "QW5pbWFsOjQ=",
                  "isCat": false,
                  "name": "Morph",
                },
              },
            ],
          },
        },
      },
    },
  },
}
`;

exports[`Functional tests subscribe connection 1`] = `
Object {
  "data": Object {
    "viewer": Object {
      "animals": Object {
        "edges": Array [
          Object {
            "node": Object {
              "name": "Felix",
            },
          },
        ],
      },
    },
  },
}
`;

exports[`Functional tests subscribe connection 2`] = `
Array [
  Array [
    "123",
    "animal",
    "QW5pbWFsOjE=",
    Object {
      "Weight": 10.1,
      "age": 6,
      "birthday": 2010-11-10T17:00:00.000Z,
      "id": 1,
      "isCat": false,
      "name": "testn",
    },
    Object {
      "subscriptionId": "123",
    },
  ],
]
`;

exports[`Functional tests subscribe connection 3`] = `Array []`;

exports[`Functional tests subscribe connection 4`] = `Array []`;

exports[`Functional tests subscribe one 1`] = `
Object {
  "data": Object {
    "viewer": Object {
      "animal": Object {
        "name": "Rex",
      },
    },
  },
}
`;

exports[`Functional tests subscribe one 2`] = `
Array [
  Array [
    "123",
    "animal",
    "QW5pbWFsOjE=",
    Object {
      "Weight": 10.1,
      "age": 6,
      "birthday": 2010-11-10T17:00:00.000Z,
      "id": 1,
      "isCat": false,
      "name": "testn",
    },
    Object {
      "subscriptionId": "123",
    },
  ],
]
`;

exports[`Functional tests update mutation: animal 1`] = `
Object {
  "data": Object {
    "updateAnimal": Object {
      "animal": Object {
        "birthday": "Wed, 07 Dec 2016 10:42:48 GMT",
        "name": "testName1",
      },
    },
  },
}
`;
